### PR TITLE
fix: refresh authentication token when necessary

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
A proposed fix for #1740

### CODE REVIEW

**Review by larsmoa — COMMENTED** [review]
Nice - but add a test :)

**Comment by haakonflatval-cognite** [comment]
> Nice - but add a test :)

I'll see if I can find a way to create one that doesn't need an hour and a half to run :P

File: viewer/packages/modeldata-api/src/CdfModelDataProvider.test.ts [hunk 1]
```diff
@@ -7,16 +7,27 @@ import { CdfModelDataProvider } from './CdfModelDataProvider';
...
   });
+
+  let authenticationSpy: jest.MockInstance<Promise<boolean>, []> = mockClientAuthentication(client);
+
   client.loginWithApiKey({ apiKey: 'dummy', project: 'unittest' });
+
   const clientExt = new CdfModelDataProvider(client);
 
+  beforeEach(() => {
+    authenticationSpy.mockRestore();
```
Comment (larsmoa) [inline viewer/packages/modeldata-api/src/CdfModelDataProvider.test.ts:27]: Move restore to `afterEach(() => { ... })`

File: viewer/packages/modeldata-api/src/CdfModelDataProvider.test.ts [hunk 2]
```diff
@@ -33,4 +44,22 @@ describe(CdfModelDataProvider.name, () => {
     const view = new Uint8Array(result);
     expect(view.toString()).toEqual(expected.toString());
   });
+
+  test('getBinaryFile() does not authenticate on 200', async () => {
+    // Make first API call fail, second succeed
+    nock(/.*/).get(/.*/).reply(200, '');
```
Comment (larsmoa) [inline viewer/packages/modeldata-api/src/CdfModelDataProvider.test.ts:50]:
```suggestion
     nock(/.*/).get(/.*/).reply(200, '');
```

File: viewer/packages/modeldata-api/src/CdfModelDataProvider.ts [hunk 3]
```diff
@@ -26,27 +28,37 @@ export class CdfModelDataProvider implements ModelDataProvider {
...
+  private async fetchWithRetry(input: RequestInfo, options: RequestInit, retries: number = 3) {
+    let error: Error | undefined;
+    for (let i = 0; i < retries; i++) {
+      try {
+        await this.authenticationPromise;
+
+        const response = await fetch(input, options);
+
+        // Authentication error
+        if (response.status == 401) {
```
Comment (larsmoa) [inline viewer/packages/modeldata-api/src/CdfModelDataProvider.ts:49]:
```suggestion
        if (response.status === 401) {
```

**Review by larsmoa — APPROVED** [review]
Looks good, just a couple of minor things

**Pull request merged**